aten/src/THCUNN/generic/MultiLabelMarginCriterion.cu

```diff
@@ -81,7 +81,7 @@ void THNN_(MultiLabelMarginCriterion_updateOutput)(
           reduction == at::Reduction::Mean
           );
       THCudaCheck(cudaGetLastError());
-      THCTensor_(set1d)(state, output, 0, ScalarConvert<accreal, scalar_t>::to(THCTensor_(sumall)(state, output_tmp)));
+      THCTensor_(set1d)(state, output, 0, ScalarConvert<accreal, scalar_t>::to(THTensor_wrap(output_tmp).sum().item<scalar_t>()));
```

Comment:
nit: probably you can call `.sum_out` writing output to `output` instead of this gymnastics with THCTensor_(set1d)? Would also avoid h2d d2h copies and synchronization. If not, then fine, it's not a regression. I don't remember if `sum_out` for the whole tensor can be persuaded to produce result on the gpu, but sum along dimension definitely can, and is just as fast.

Reply:
I don't think there is:
![image](https://user-images.githubusercontent.com/1032377/82990748-83f32800-9fb1-11ea-90a5-b9ba7b8720f6.png)

Reply:
sum.IntList_out should do the trick?

Reply:
trying